operator/internal/handlers/lokistack_create_or_update.go

```diff
@@ -240,6 +243,35 @@ func CreateOrUpdateLokiStack(
 		}
 	}
 
+	var tlsProfile openshiftv1.TLSSecurityProfile
+
+	if opts.TLSProfileType != "" {
+		tlsProfile = openshiftv1.TLSSecurityProfile{
+			Type: openshiftv1.TLSProfileType(opts.TLSProfileType),
+		}
+	} else {
+		tlsProfile = openshiftv1.TLSSecurityProfile{
+			Type: openshiftv1.TLSProfileIntermediateType,
+		}
+
+		var apiServer openshiftv1.APIServer
+		if err = k.Get(ctx, client.ObjectKey{Name: "cluster"}, &apiServer); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return kverrors.Wrap(err, "failed to lookup apiServer")
+			}
+		}
+
+		if apiServer.Spec.TLSSecurityProfile != nil {
+			tlsProfile = *apiServer.Spec.TLSSecurityProfile
+		}
+	}
+
+	tlsMinVersion, ciphers := getSecurityProfileInfo(&tlsProfile)
+	opts.TLSProfileSpec = manifests.TLSProfileSpec{
+		MinTLSVersion: tlsMinVersion,
+		Ciphers:       ciphers,
+	}
```

Comment:
Can we extract this in a separate file under `operator/internal/handlers/internal/tlsprofile` like we do for rules and other companion objects?